web/libs/editor/src/components/InteractiveOverlays/CommentsOverlay.module.scss

```diff
@@ -0,0 +1,53 @@
+$icon-primary-color: var(--grape_600);
+$icon-secondary-color: var(--grape_0);
+
+.container {
+  position: absolute;
+  top: 0;
+  left: 0;
+  width: 100%;
+  height: 100%;
+  pointer-events: none;
+  z-index: 100;
+
+  .commentItem {
+    transition: opacity 300ms;
+    transform: translateY(-100%);
+    filter: drop-shadow(0 2px 8px #12110D52) drop-shadow(0 8px 16px #12110D1F);
+    pointer-events: all;
+    cursor: pointer;
+  }
+
+  .commentIcon {
+    /* for the future */
+  }
+
+  .commentIconBackground {
+    transition: fill 300ms;
+    fill: $icon-secondary-color;
+  }
+
+  .commentIconLines {
+    transition: fill 300ms;
+    fill: $icon-primary-color;
+  }
+
+  &._highlighting {
+    & .commentItem {
+      opacity: 0.2;
+
+      &._highlighted {
+        opacity: 1;
+
+        & .commentIconBackground {
+          fill: $icon-primary-color;
+        }
+
+        & .commentIconLines {
+          fill: $icon-secondary-color;
+        }
+      }
+    }
+  }
```

Comment:
Prefer css variables to SCSS variables, as the latter modifies the underlying value at compile time and leaves a static hex value, and breaks the ability to readily theme.

```suggestion
.container {
  --icon-primary-color: var(--grape_600);
  --icon-secondary-color: var(--grape_0);
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  pointer-events: none;
  z-index: 100;

  .commentItem {
    transition: opacity 300ms;
    transform: translateY(-100%);
    filter: drop-shadow(0 2px 8px #12110D52) drop-shadow(0 8px 16px #12110D1F);
    pointer-events: all;
    cursor: pointer;
  }

  .commentIcon {
    /* for the future */
  }

  .commentIconBackground {
    transition: fill 300ms;
    fill: var(--icon-secondary-color);
  }

  .commentIconLines {
    transition: fill 300ms;
    fill: var(--icon-primary-color);
  }

  &._highlighting {
    & .commentItem {
      opacity: 0.2;

      &._highlighted {
        opacity: 1;

        & .commentIconBackground {
          fill: var(--icon-primary-color);
        }

        & .commentIconLines {
          fill: var(--icon-secondary-color);
        }
      }
    }
  }
```

Reply:
we have (`--sand_900`) but we dont have a css variable that does `sand_900` with `52` or `1F` as opacity
also since sand_900 is in hex like we have it here we can just stick it into rgb or rgba - we should make sure to update those color css variables to be rgb or rgba when we move further into the css tokenization to help simplify this sort of situation.
in the mean time maybe we can look at using something like the [opacify](https://sass-lang.com/documentation/modules/color/#opacify) sass function

Reply:
Further up I mentioned how we should avoid SASS level functions, as anything done through SASS is compile-time only and cannot be easily modified through swapping our css variables (theming). I think if we don't have an immediate answer that is totally fine, just thought to check, but given that I would say we leave this as is for now. It can be found and replaced with a proper solution in a dedicated PR at later time.

Reply:
that makes sense - but defintly something to follow up with as part of the upcoming tokens initiative

Reply:
Just to be clear, the part I am mentioning keeping as is will be just this line:

```
filter: drop-shadow(0 2px 8px #12110D52) drop-shadow(0 8px 16px #12110D1F);
```

The suggestion I have above still stands.